test/Concurrency/Runtime/exclusivity.swift

```diff
@@ -0,0 +1,42 @@
+// RUN: %target-run-simple-swift(-Xfrontend -enable-experimental-concurrency  %import-libdispatch -parse-as-library)
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: libdispatch
+
+// rdar://76038845
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: back_deployment_runtime
+
+// This test makes sure that:
+//
+// 1. Tasks have separate exclusivity sets.
+// 2. Exercise the pushing/popping of access sets from tasks.
+
+import _Concurrency
+
+var global: Int = 5
+
+@available(SwiftStdlib 5.5, *)
+@main
+struct Runner {
+    // Tasks today have separate exclusivity access sets so there shouldn't be an
+    // exclusivity violation here.
+    static func myMain(_ x: inout Int, _ shouldSleep: Bool) async {
+      print("In my main")
+    }
+
+    static func myMain2(_ x: inout Int, _ shouldSleep: Bool) async {
```

Comment:
I am going to add some more tests that exercise this where there are multiple elements in the exclusivity set we need to move and also the synchronous case we talked about.